src/App.js

```diff
@@ -1,36 +1,59 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect } from 'react';
+import detectEthereumProvider from '@metamask/detect-provider'
 import './App.css';
-import { Web3Provider } from 'react-web3';
 import AddTokenPanel from './AddTokenPanel';
 import EditTokenPanel from './EditTokenPanel';
 import DownloadMetaMaskButton from './DownloadMetaMaskButton';
 import { BrowserRouter, Route, Switch, HashRouter } from 'react-router-dom';
 
-class App extends Component {
-
-  render(props, context) {
-    return (
-      <div className="App">
-        <a className="github-banner" href="https://github.com/MetaMask/Add-Token"><img src="https://s3.amazonaws.com/github/ribbons/forkme_right_gray_6d6d6d.png" alt="Fork me on GitHub" /></a>
-        <Web3Provider
-          web3UnavailableScreen={() => <div>
-            <p>You need a web3 browser like MetaMask to use this site and manage cryptocurrencies.</p>
-            <DownloadMetaMaskButton/>
-          </div>}
-        >
-          <BrowserRouter basename={process.env.PUBLIC_URL}>
-            <HashRouter hashType="noslash">
-              <Switch>
-                <Route path="/edit" component={EditTokenPanel} />
-                <Route path="/add" component={AddTokenPanel} />
-                <Route path="/" component={AddTokenPanel} />
-              </Switch>
-            </HashRouter>
-          </BrowserRouter>
-        </Web3Provider>
-      </div>
-    );
+
+const MainContent = () => {
+  return (
+    <div>
+      <BrowserRouter basename={process.env.PUBLIC_URL}>
+        <HashRouter hashType="noslash">
+          <Switch>
+            <Route path="/edit" component={EditTokenPanel} />
+            <Route path="/add" component={AddTokenPanel} />
+            <Route path="/" component={AddTokenPanel} />
+          </Switch>
+        </HashRouter>
+      </BrowserRouter>
+    </div>
+  )
+};
+
+const ErrorContent = () => {
+  return (
+    <div>
+      <p>You need a web3 browser like MetaMask to use this site and manage cryptocurrencies.</p>
+      <DownloadMetaMaskButton />
+    </div>
+  )
+}
+
+const App = () => {
+  const [isProviderLoaded, setProvider] = useState([]);
+
+  const checkEthereumProvider = async () => {
+    const provider = await detectEthereumProvider()
+    setProvider(provider)
   }
+
+  useEffect(() => {
+    checkEthereumProvider()
+  }, [isProviderLoaded])
```

Comment:
Why is `isProviderLoaded` here? Putting `isProviderLoaded` here will result in this being run again when `isProviderLoaded` changes. It will change after it's run for the first time, so this will run twice each time for no reason.

The `useEffect` dependencies should include any values used by the effect that can change over time. `isProviderLoaded` is _set_ by the effect but it's not used by it. This effect only relies upon static functions (`detectEthereumProvider` and `setProvider`), so it doesn't really need any dependencies.

It also uses the helper function `checkEthereumProvider`, but I'd strongly advise moving that inside of `useEffect` rather than it being a separate function that gets declared each render. This pattern of declaring helper functions separately that are solely for use within the effect make it more difficult to see what the effect's dependencies are. See this doc for more info: https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies